### PR TITLE
feat(api): explicit setup/doctor modes + v5.4.0 contract audit

### PR DESCRIPTION
**Pull request by ndycode**
### Summary
**Overall**: APPROVED
**Breaking Changes**: NONE

This PR implements a public API contract hardening pass for `v5.3.4..HEAD` and adds compatibility-focused guardrails/documentation.

### Breaking Changes Found
- None detected in exported TypeScript signatures for touched public surfaces.
- None detected in existing `codex-*` tool names or existing boolean argument compatibility.

### API Design Issues
- `index.ts` (`codex-setup`, `codex-doctor`) previously relied on boolean-only mode selection (`wizard`, `deep`, `fix`) which is ambiguous for callers and scripts.
- Addressed by adding additive explicit `mode` arguments:
  - `codex-setup mode="checklist|wizard"`
  - `codex-doctor mode="standard|deep|fix"`
- Legacy booleans remain supported for backward compatibility.
- Conflict detection now returns explicit contract errors when `mode` and booleans disagree.

### Error Contract Issues
- Added explicit invalid-mode and conflicting-option error behavior for both tools.
- Documented changed error semantics and compatibility guidance in:
  - `docs/development/API_CONTRACT_AUDIT_v5.4.0.md`

### Versioning Recommendation
**Suggested bump**: MINOR
**Rationale**: additive caller-visible argument capability (`mode`) with full backward compatibility retained for legacy flags.

## Migration Path
Existing callers do not need to change.

Recommended updates for clarity:
- `codex-setup wizard=true` -> `codex-setup mode="wizard"`
- `codex-doctor deep=true` -> `codex-doctor mode="deep"`
- `codex-doctor fix=true` -> `codex-doctor mode="fix"`

## Validation
- `npm run typecheck`
- `npm test`
- `npm run build`

All passed in isolated worktree branch `audit/api-contract-v5-4-0`.

## Summary by CodeRabbit

* **New Features**
  * Introduced mode-based command syntax (mode="wizard", mode="deep", mode="fix", mode="checklist", mode="standard") as preferred alternatives to boolean flags.
  * Maintained backward compatibility with legacy boolean flags.

* **Documentation**
  * Updated command examples across guides to reflect new mode-based syntax.
  * Added API Contract Audit documentation for v5.4.0.

**note:** greptile review for oc-chatgpt-multi-auth. cite files like `lib/foo.ts:123`. confirm regression tests + windows concurrency/token redaction coverage.
---

<h3>Greptile Summary</h3>

adds explicit `mode` parameters to `codex-setup` and `codex-doctor` commands while preserving backward compatibility with legacy boolean flags. implements robust validation (empty strings, whitespace, invalid values) and conflict detection between new mode syntax and legacy flags.

**key changes:**
- `codex-setup` now accepts `mode="checklist|wizard"` (preferred) or legacy `wizard=true`
- `codex-doctor` now accepts `mode="standard|deep|fix"` (preferred) or legacy `deep=true`/`fix=true`
- validation rejects empty/whitespace/invalid mode values with clear error messages
- conflict detection prevents contradictory inputs like `mode="checklist" wizard=true`
- comprehensive vitest coverage added for all edge cases (empty strings, conflicts, backward compat)
- all user-facing docs and error messages updated to new syntax

**safety assessment:**
no filesystem operations, token handling, logging changes, or concurrency introduced. purely input validation and api contract clarification. does not touch windows filesystem paths, token storage (`lib/storage.ts`), refresh queues, or audit logs.

<h3>Confidence Score: 5/5</h3>

- safe to merge with zero risk - purely additive api improvements with full backward compatibility
- validation logic is sound (handles empty/whitespace/invalid inputs), conflict detection prevents ambiguous states, comprehensive test coverage including edge cases, no runtime behavior changes to filesystem/token/logging subsystems, documentation consistently updated across all files
- no files require special attention

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| index.ts | adds explicit mode parameters to codex-setup and codex-doctor with robust validation, conflict detection, and backward compatibility for legacy boolean flags |
| test/index.test.ts | comprehensive test coverage for new mode parameters including edge cases (empty strings, whitespace, invalid values, conflicts) |
| docs/development/API_CONTRACT_AUDIT_v5.4.0.md | new audit documentation explaining api contract stability, compatibility matrix, and migration guidance for v5.4.0 |
| CHANGELOG.md | changelog entries for explicit mode syntax and api contract audit docs |
| README.md | updated command examples to use mode syntax with backward compatibility notes |

</details>

<sub>Last reviewed commit: c651eec</sub>

### CODE REVIEW

**Comment by ndycode** [comment]
Deep-audit update pushed with additional atomic commits.

What was added in this pass:
- `test(api): expand mode contract compatibility coverage`
  - Added explicit coverage for `codex-setup mode="checklist"`
  - Added explicit coverage for `codex-doctor mode="standard"`
  - Added explicit coverage for `codex-doctor mode="fix"`
  - Existing legacy boolean compatibility tests retained
- `docs(api): deepen v5.4 contract matrix and compatibility evidence`
  - Replaced audit doc with full methodology, export diff matrix, tool inventory diff, file-by-file classification, error contract matrix, migration guidance, and semver rationale

Verification rerun after these commits:
- `npm run typecheck` PASS
- `npm test` PASS (`56` files, `1785` tests)
- `npm run build` PASS

File: index.ts [hunk 1]
```diff
@@ -4495,23 +4495,41 @@ while (attempted.size < Math.max(1, accountCount)) {
...
 						.optional()
-						.describe("Launch menu-driven setup wizard when terminal supports it."),
+						.describe("Legacy alias for mode=\"wizard\" (backward compatible)."),
 				},
-				async execute({ wizard }: { wizard?: boolean } = {}) {
+				async execute({ mode, wizard }: { mode?: string; wizard?: boolean } = {}) {
+					const normalizedMode = mode?.trim().toLowerCase();
+					if (normalizedMode && normalizedMode !== "checklist" && normalizedMode !== "wizard") {
+						return `Invalid mode: ${mode}\n\nValid modes: checklist, wizard`;
+					}
```
[the remark this answers is not in the document]
Comment (ndycode) [reply index.ts:4522]: Implemented in commit c651eec. Empty/whitespace mode inputs are now rejected for both `codex-setup` and `codex-doctor` using explicit `mode !== undefined && !normalizedMode` guards. Added regression tests in `test/index.test.ts` for `mode=""` and whitespace values for both tools. Validation rerun: typecheck/test/build all pass.

**Comment by ndycode** [comment]
Addressed all actionable review comments.

Latest fix: `c651eec`
- Reject empty/whitespace mode values for `codex-setup` and `codex-doctor`
- Added regression tests for empty/whitespace modes
- Revalidated with:
  - `npm run typecheck`
  - `npm test`
  - `npm run build`

All review threads are now resolved.

**Comment by ndycode** [comment]
@coderabbitai review

**Comment by ndycode** [comment]
@greptile please review

**Comment by ndycode** [comment]
Deep comment-audit summary (2026-02-28)

Audited all current PR comments for **#43** (feat(api): explicit setup/doctor modes + v5.4.0 contract audit) including resolved/outdated review threads.
Counts: issue comments = **6**, review threads = **1**, review-thread comments = **3**.
No new code change was required for this PR based on current comment text and branch head state.

Per-comment disposition ledger:

- https://github.com/ndycode/oc-chatgpt-multi-auth/pull/43#issuecomment-3975745764 -> N/A (bot meta/status comment; no code-change request)
- https://github.com/ndycode/oc-chatgpt-multi-auth/pull/43#issuecomment-3975759505 -> N/A (informational/progress comment; no new actionable request)
- https://github.com/ndycode/oc-chatgpt-multi-auth/pull/43#issuecomment-3975816231 -> N/A (informational/progress comment; no new actionable request)
- https://github.com/ndycode/oc-chatgpt-multi-auth/pull/43#issuecomment-3975925752 -> N/A (manual review trigger only)
- https://github.com/ndycode/oc-chatgpt-multi-auth/pull/43#issuecomment-3975926034 -> N/A (bot meta/status comment; no code-change request)
- https://github.com/ndycode/oc-chatgpt-multi-auth/pull/43#issuecomment-3976272224 -> N/A (manual review trigger only)
- https://github.com/ndycode/oc-chatgpt-multi-auth/pull/43#discussion_r2866675306 -> Already satisfied at current HEAD (`c651eec`); empty/whitespace mode now rejects for setup/doctor (`index.ts:4516-4521`, `4558-4564`).
- https://github.com/ndycode/oc-chatgpt-multi-auth/pull/43#discussion_r2866712406 -> N/A (author status/progress note)
- https://github.com/ndycode/oc-chatgpt-multi-auth/pull/43#discussion_r2866713164 -> N/A (bot acknowledgement/meta)

**Comment by ndycode** [comment]
@greptile please review latest updates

**Comment by ndycode** [comment]
Superseded by #53, which consolidates and deep-audits all open PR changes.
